ipn/ipnlocal/network-lock.go

```diff
@@ -237,3 +348,50 @@ func (b *LocalBackend) tkaInitFinish(nm *netmap.NetworkMap, nks map[tailcfg.Node
 		return a, nil
 	}
 }
+
+// tkaFetchBootstrap sends a /machine/tka/bootstrap RPC to the control plane
+// over noise. This is used to get values necessary to enable or disable TKA.
+func (b *LocalBackend) tkaFetchBootstrap(nodeID tailcfg.NodeID, head tka.AUMHash) (*tailcfg.TKABootstrapResponse, error) {
+	bootstrapReq := tailcfg.TKABootstrapRequest{
+		NodeID: nodeID,
+	}
+	if head != (tka.AUMHash{}) {
+		head, err := head.MarshalText()
+		if err != nil {
+			return nil, fmt.Errorf("head.MarshalText failed: %v", err)
+		}
+		bootstrapReq.Head = string(head)
+	}
+
+	var req bytes.Buffer
+	if err := json.NewEncoder(&req).Encode(bootstrapReq); err != nil {
+		return nil, fmt.Errorf("encoding request: %v", err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	if err := ctx.Err(); err != nil {
+		return nil, fmt.Errorf("ctx: %w", err)
+	}
+	req2, err := http.NewRequestWithContext(ctx, "GET", "https://example.com/machine/tka/bootstrap", &req)
```

Comment:
is example.com what we do elsewhere? How about just "control"? or "control-noise" ?

Reply:
I've seen `https://example.com` in some places and `https://unused` in others, the latter maybe being slightly more prevalent.

Reply:
unused is better than example.com which looks like it's docs

Reply:
`httptest` will only accept `example.com` unfortunately.

Reply:
I'd suggest using the control URL then, and tests can override that?

Reply:
That unfortunately won't work either, as the TLS config, certificate, and key are quite literally hardcoded in Go: https://cs.opensource.google/go/go/+/refs/tags/go1.19.1:src/net/http/internal/testcert/testcert.go;l=14;drc=d8220ac32fc73e69aa6d58890b6b5277d277825f

We could do the work to generate a fake config, cert, and key, but I'd rather not because I like that `example.com` indicates the request.Do() flow is tested somewhere whereas `unused` suggests theres no such test.

Reply:
Oh sorry, I was jumping a bunch without explaining my thought further. If the request is made to the control URL, that makes this code clearer. To make testing with `httptest` work correctly, set the control URL to `example.com` when initializing a test. Would that satisfy everything?

Reply:
I'm not sure I follow. Both the noise and test http clients use custom transports, which wire up their own connections and hence dont look at the host part of the URL at all. The only thing that looks at the host part is the TLS verification logic, which (because the server cert is hardcoded to `example.com`) will fail for any other domain.

Reply:
Actually I just figured out a way to get to the client TLS config from the http.Client, meaning I can disable verification and we can just used `unused`.